main.cpp

```diff
@@ -92,27 +109,79 @@ void link_executable(const Buildfile& buildfile, const vector<string>& cpp_files
     }
 }
 
-int main() {
+bool is_created(const string &filename){
+
+    // tests for the existence of the file with success being a 0
+    if(access(filename.c_str(), F_OK) == 0)
+    {
+        return true;
+    }
+    return false;
+}
+
+bool parse_commandline_args(int argc, char** &argv)
+{
+    struct option longOpts[] = {
+            { "help", no_argument, nullptr, 'h' },
+            { nullptr, 0, nullptr, '\0' }
+        };
+
+    int option, optionIndex;
+
+    while((option = getopt_long(argc, argv, "h", longOpts, &optionIndex))!= -1)
+    {
+        switch(option)
+        {
+            case 'h':
+            {
+                cout<<help_header;
+                return false;
+            }
+
+            default:
+            {
+                return false;
+                break;
+            }
+        }
+    }
+    return true;
+}
+
+int main(int argc, char ** argv) {
     /*
-     * 1) get the cpp files
-     * 2) determine what cpp files to recompile
+     * 1) parse for commandline arguments
+     * 2) get the cpp files
+     * 3) determine what cpp files to recompile
      *      - need to recompile if no obj file we have to create it, or if there is an obj file,
      *      but the cpp file has been modified more recently than the obj file
-     * 3) if none, we are done
-     * 4) else, recompile them (done!)
-     * 5) relink everything
-     * 6) done!
+     *      - need to check to see if there is an executable created
+     * 4) if none, we are done
+     * 5) else, recompile them (done!)
+     * 6) relink everything
+     * 7) done!
      */
     try {
+
+        /*
+        parse command line arguments, if we get false back we encountered either a help or invalid
+        option and should exit the program while displaying the help message.
+        */
+        if(parse_commandline_args(argc, argv) == false)
+        {
+            cout << help_msg;
+            return 0;
```

Comment:
See above comment about how this is buggy- will return a 0 (OK) status code even when an invalid flag was provided